analyzere/base_resources.py

```diff
@@ -148,6 +149,13 @@ def to_dict(self):
             d[k] = to_dict(v)
         return d
 
+    def __hash__(self):
+        # Certainly not the most efficient hash, since anything without an ID
+        # has the same hash value.
+        if hasattr(self, "id"):
+            int(uuid.UUID(self.id))
+        return 0
```

Comment:
@pmdarrow I think you're right, in that we could do more (it's a pretty terrible hash function), but we can always improve upon it later. Hash functions don't need to be consistent across instances of python, so we can improve later (if it becomes a problem) and not break anything. This was just quick and "technically correct".